coherent_rates/system.py

```diff
@@ -72,128 +70,276 @@
 _L0Inv = TypeVar("_L0Inv", bound=int)
 
 
+def _get_fundamental_potential_1d(
+    system: PeriodicSystem1D,
+) -> Potential[TupleBasis[FundamentalTransformedPositionBasis1d[Literal[3]]]]:
+    """Generate potential for a periodic 1D system."""
+    delta_x = np.sqrt(3) * system.lattice_constant / 2
+    axis = FundamentalTransformedPositionBasis1d[Literal[3]](np.array([delta_x]), 3)
+    vector = 0.25 * system.barrier_energy * np.array([2, -1, -1]) * np.sqrt(3)
+    return {"basis": TupleBasis(axis), "data": vector}
+
+
+def _get_fundamental_potential_2d(
+    system: PeriodicSystem2D,
+) -> Potential[
+    TupleBasis[
+        FundamentalTransformedPositionBasis[Literal[3], Literal[2]],
+        FundamentalTransformedPositionBasis[Literal[3], Literal[2]],
+    ]
+]:
+    # We want the simplest possible potential in 2d with symmetry
+    # (x0,x1) -> (x1,x0)
+    # (x0,x1) -> (-x0,x1)
+    # (x0,x1) -> (x0,-x1)
+    # We therefore occupy G = +-K0, +-K1, +-(K0+K1) equally
+    data = [[0, 1, 1], [1, 1, 0], [1, 0, 1]]
+    vector = 0.5 * system.barrier_energy * np.array(data) / np.sqrt(9)
+    return {
+        "basis": TupleBasis(
+            FundamentalTransformedPositionBasis[Literal[3], Literal[2]](
+                system.lattice_constant * np.array([0, 1]),
+                3,
+            ),
+            FundamentalTransformedPositionBasis[Literal[3], Literal[2]](
+                system.lattice_constant
+                * np.array(
+                    [np.sin(np.pi / 3), np.cos(np.pi / 3)],
+                ),
+                3,
+            ),
+        ),
+        "data": vector.ravel(),
+    }
+
+
 @dataclass
 class PeriodicSystem:
-    """Represents the properties of a 1D Periodic System."""
+    """Represents the properties of a Periodic System."""
 
     id: str
     """A unique ID, for use in caching"""
     barrier_energy: float
     lattice_constant: float
     mass: float
 
+    def get_fundamental_potential(self: Self) -> Potential[Any]:
+        ...
+
+    def potential(
```

Comment:
if it is a function it should be named after what it does. In this case it is get_potential